src/ert/ensemble_evaluator/monitor_async.py

```diff
@@ -0,0 +1,135 @@
+import asyncio
+import logging
+import pickle
+import ssl
+import uuid
+from typing import TYPE_CHECKING, Any, AsyncGenerator, Optional
+
+from cloudevents.conversion import to_json
+from cloudevents.exceptions import DataUnmarshallerError
+from cloudevents.http import CloudEvent, from_json
+from websockets import Headers, WebSocketClientProtocol
+from websockets.client import connect
+
+from ert.ensemble_evaluator import identifiers
+from ert.ensemble_evaluator._wait_for_evaluator import wait_for_evaluator
+from ert.serialization import evaluator_marshaller, evaluator_unmarshaller
+
+if TYPE_CHECKING:
+    from ert.ensemble_evaluator.evaluator_connection_info import EvaluatorConnectionInfo
+
+
+logger = logging.getLogger(__name__)
+
+
+class MonitorAsync:
+    def __init__(self, ee_con_info: "EvaluatorConnectionInfo") -> None:
```

Comment:
Should we add `from future import annotations` in the import list, and remove the quotes for the typing?

Reply:
I prefer the quotes tbh, as it prevents extra imports. :shrug:

Reply:
I agree with @jonathan-eq, can put the import under `if TYPE_CHECKING`.